src/Bicep.Core/TypeSystem/OciTypeLoader.cs

```diff
@@ -18,7 +23,32 @@ private OciTypeLoader(ImmutableDictionary<string, byte[]> typesCache)
             this.typesCache = typesCache;
         }
 
-        public static OciTypeLoader FromTgz(Stream stream)
+        private static OciTypeLoader FromDiskHelper(IFileSystem fs, Uri? typesTgzUri)
+        {
+            if (typesTgzUri is null)
+            {
+                throw new ArgumentNullException(nameof(typesTgzUri));
+            }
+
+            using var stream = fs.File.OpenRead(typesTgzUri.LocalPath);
+
+            return FromStream(stream);
+        }
+
+        public static OciTypeLoader FromDisk(IFileSystem fs, Uri? typesTgzUri)
```

Comment:
Let's make this non-null and handle the possibility of it being null in the caller method.